src/main/java/net/wurstclient/hacks/AutoLightingHack.java

```diff
@@ -0,0 +1,476 @@
+
+        return true;
+    }
+
+    @Override
+    protected void onEnable() {
+        EVENTS.add(UpdateListener.class, this);
+        EVENTS.add(RenderListener.class, this);
+        this.status = Status.Idle;
+    }
+
+    @Override
+    protected void onDisable() {
+        EVENTS.remove(UpdateListener.class, this);
+        EVENTS.remove(RenderListener.class, this);
+        if (this.currentSearchTask != null) this.currentSearchTask.cancel(true);
+        this.currentSearchTask = null;
+    }
+
+    @Override
+    public void onUpdate() {
+        if (this.currentSearchTask != null && !this.currentSearchTask.isDone()) {
+            this.status = Status.Searching;
+            return;
+        }
+        if (this.idleCooldown > 0) {
+            this.idleCooldown--;
+            this.status = Status.Idle;
+            return;
+        }
+
+        if (this.currentSearchTask != null) {
+            try {
+                this.putPos = this.currentSearchTask.get();
+            } catch (Exception e) {
+                e.printStackTrace();
+                ChatUtils.error("[" + super.getName() + "] Can not search: " + e);
+                super.setEnabled(false);
+            }
+            this.currentSearchTask = null;
+        }
+
+        if (this.putPos == null) this.currentSearchTask = this.executorService.submit(this::search);
+
+        if (placeBlock()) this.putPos = null;
+    }
+
+    private Vec3d ofBottom(BlockPos place) {
+        return Vec3d.add(place, 0.5, 0, 0.5);
+    }
+
+    private @Nullable BlockPos search() {
+        final int level = 14 - 1;
+        final double range1 = this.range1.getValue();
+        final double range1Sq = MathHelper.square(range1);
+        final double range2Sq = this.range2.getValueSq();
+        final double range3Sq = this.range3.getValueSq();
+        final double frontpath = this.frontpath.getValue();
+        final double frontpathSq = MathHelper.square(frontpath);
+        Vec3d eyesVec = RotationUtils.getEyesPos();
+        ClientWorld world = Objects.requireNonNull(MC.world);
+        @SuppressWarnings("all") Stream<BlockPos> stream =//
+            BlockUtils.getAllInBoxStream(BlockPos.ofFloored(eyesVec), range3.getValueCeil())//
+                .filter(pos -> eyesVec.squaredDistanceTo(Vec3d.ofCenter(pos)) <= range3Sq)//
+                .sorted(Comparator.comparingDouble(pos -> eyesVec.squaredDistanceTo(Vec3d.ofCenter(pos))))//
+                .filter(pos -> {
+                    BlockState state = world.getBlockState(pos);
+
+                    if (state.blocksMovement()) return false;
+                    if (!state.getFluidState().isEmpty()) return false;
+
+                    BlockState stateDown = world.getBlockState(pos.down());
+                    if (!stateDown.allowsSpawning(world, pos.down(), EntityType.ZOMBIE)) return false;
+
+                    return world.getLightLevel(LightType.BLOCK, pos) < 1;
+                })//
+            ;

```

Comment:
I'm not sure if the sorted order is used somewhere later, but tbh I'm more concerned about the `@SuppressWarnings("all")` and the strange formatting of this code with a bunch of empty comments after some lines. Can you tell what warning is being suppressed here?

Reply:
I agree with your suggestion. I would also add that, if the `@SuppressWarnings` annotation is necessary for the code to compile, then it would better to suppress the specific warning that appears on this code instead of suppressing all warnings. It would also be a good idea to add a comment explaining why the warning suppression is necessary.

As for the empty comments, they do not align with the project's formatting style and should be removed.